src/subdomains/core/liquidity-management/adapters/actions/scrypt.adapter.ts

```diff
@@ -0,0 +1,158 @@
+  }
+
+  async checkCompletion(order: LiquidityManagementOrder): Promise<boolean> {
+    switch (order.action.command) {
+      case ScryptAdapterCommands.WITHDRAW:
+        return this.checkWithdrawCompletion(order);
+
+      default:
+        return false;
+    }
+  }
+
+  validateParams(command: string, params: Record<string, unknown>): boolean {
+    switch (command) {
+      case ScryptAdapterCommands.WITHDRAW:
+        return this.validateWithdrawParams(params);
+
+      default:
+        throw new Error(`Command ${command} not supported by ScryptAdapter`);
+    }
+  }
+
+  // --- COMMAND IMPLEMENTATIONS --- //
+
+  private async withdraw(order: LiquidityManagementOrder): Promise<CorrelationId> {
+    const { address, asset, blockchain } = this.parseWithdrawParams(order.action.paramMap);
+
+    const token = asset ?? order.pipeline.rule.targetAsset.dexName;
+
+    const balance = await this.scryptService.getAvailableBalance(token);
+    if (order.minAmount > balance) {
+      throw new OrderNotProcessableException(
+        `Scrypt: not enough balance for ${token} (balance: ${balance}, min. requested: ${order.minAmount}, max. requested: ${order.maxAmount})`,
+      );
+    }
+
+    const amount = Util.floor(Math.min(order.maxAmount, balance), 6);
+
+    order.inputAmount = amount;
+    order.inputAsset = token;
+    order.outputAsset = token;
+
+    try {
+      const response = await this.scryptService.withdrawFunds(token, amount, address);
+
+      return response.id;
+    } catch (e) {
+      if (this.isBalanceTooLowError(e)) {
+        throw new OrderNotProcessableException(e.message);
+      }
+
+      throw e;
+    }
+  }
+
+  // --- COMPLETION CHECKS --- //
+
+  private async checkWithdrawCompletion(order: LiquidityManagementOrder): Promise<boolean> {
+    const {
+      pipeline: {
+        rule: { targetAsset },
+      },
+      action: { paramMap },
+      correlationId,
+    } = order;
+
+    const { asset, blockchain } = this.parseWithdrawParams(paramMap);
+    const token = asset ?? targetAsset.dexName;
+
+    const withdrawal = await this.scryptService.getWithdrawalStatus(correlationId);
+    if (!withdrawal) {
+      this.logger.verbose(`No withdrawal for clReqId ${correlationId} at Scrypt found`);
+      return false;
+    }
+
+    if (withdrawal.status === 'Failed' || withdrawal.status === 'Rejected') {
+      throw new OrderFailedException(`Withdrawal ${correlationId} has failed with status: ${withdrawal.status}`);
+    }
+
+    if (withdrawal.status !== 'Complete') {
+      this.logger.verbose(`Withdrawal ${correlationId} status: ${withdrawal.status}`);
+      return false;
+    }
+
+    order.outputAmount = withdrawal.amount ?? order.inputAmount;
+
+    // For crypto withdrawals, verify on-chain completion
+    if (blockchain) {
+      // Scrypt doesn't return txid, so we consider Complete status as done
+      return true;
+    }
+
+    return true;

```

Comment:
- If withdraw to exchange, we need to verify deposit on target exchange
- For withdraw to address I'm not sure 🤔